packages/abi-coder/src/encoding/coders/v1/EnumCoder.test.ts

```diff
@@ -46,6 +55,12 @@ describe('EnumCoder', () => {
     expect(actualLength).toBe(expectedLength);
   });
 
+  it('should throw an error when encoding if no enum key is provided', () => {
+    const invalidCoder = new EnumCoder('TestEnum', {});
+
+    expect(() => invalidCoder.encode({} as never)).toThrow('A field for the case must be provided');
```

Comment:
[6b0ea22](https://github.com/FuelLabs/fuels-ts/pull/1780/commits/6b0ea2211b79d0a2bcef9a7f40196ef0ca0fcfb4)

Reply:
Cool! It'd be great if you could share your opinion later on.

More precisely, if its usage simplifies things or not.

Reply:
Ah I get you.

I am in favour of moving these over but I wouldn't say simplification is what we've gained from using `expectToThrowError`. In fact I'd say chaining `toThrow` to an `expect` is more familiar and therefore simpler regarding what you'd expect when testing these kind of cases.

However `expectToThrowFuelError` is stricter and therefore should give us greater confidence in our `throw` test scenarios. Since we have a custom error class it makes sense to match on that exactly rather than just a message. `expect().toThrow()` also allows partial matches on error messages, so the test writer could be lazy and the test will still pass.

Reply:
I agree, but I asked for your input because I have mixed feelings about it. 😄